### ISSUE TITLE
Ability to provide custom context message

### ISSUE THREAD

**Issue by vorner** [yoshuawuyts/context-attribute#4]
# Feature Request
## Summary

Allow the attribute to have the form of `#[context(msg = "Failed to perform operation...")]` as an alternative to use the doc comment.

## Motivation

Oftentimes, the doc comment is long and describes what the function *does*, eg „Performs operation XYZ“.

On the other hand, if one wants the error message to end up in logs or somewhere the user wants to read it, it should say what *didn't* happen ‒ „Failed to perform operation XYZ“.

If one wants both sane documentation on the functions and nice error messages in logs, there's contention about what the doc comment should contain.

This thing would allow writing two different messages, one for documentation, another for the error while still saving the boilerplate to call `.context()` on the caller side.

It also allows future extension of formatting the context using the parameters, but that would be the next steps.

(This is just throwing the raw idea in, I'm not filling the rest of the sections yet as they are IMO both premature and quite obvious).

**Comment by woodgear** [comment]
you might be interested in this #5

**Comment by vorner** [comment]
Yes, that looks quite nice (I've looked at the examples, not at the code much).
